v2/internal/testcommon/test_context.go

```diff
@@ -87,11 +88,11 @@ func NewTestContext(
 	}
 }
 
-func (tc TestContext) ForTest(t *testing.T, cfg config.Values) (PerTestContext, error) {
+func (tc TestContext) ForTest(t *testing.T, cfg config.Values, hideCustomData map[string]string) (PerTestContext, error) {
```

Comment:
Using a map here makes this complex to set up when it is needed.

Why not keep the original signature for the factory method, but add a `WithRedaction(regex string, placeholder string)` method that can be used to add any required additional rules. It'd return the `TestContext` to allow chaining.